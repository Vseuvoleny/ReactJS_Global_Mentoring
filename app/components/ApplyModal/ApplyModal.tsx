import React, { memo } from "react";
import MovieForm from "../MovieForm/MovieForm";
import classes from "./ApplyModal.module.scss";

const ApplyModal = memo(() => {
  return (
    <section className={classes.apply_modal}>
      <span className={classes.apply_modal__title}>add movie</span>
      <MovieForm />
    </section>
  );
});

export default ApplyModal;
