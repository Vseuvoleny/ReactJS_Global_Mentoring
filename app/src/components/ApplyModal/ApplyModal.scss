.apply-modal {
  &__title {
    font-size: 40px;
    line-height: 49px;
    letter-spacing: 1px;
    text-transform: uppercase;
  }
}
