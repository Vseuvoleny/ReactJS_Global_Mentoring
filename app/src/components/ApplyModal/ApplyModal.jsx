import React from "react";
import MovieForm from "../MovieForm/MovieForm";
import "./ApplyModal.scss";

const ApplyModal = () => {
  return (
    <section className="apply-modal">
      <span className="apply-modal__title">add movie</span>
      <MovieForm />
    </section>
  );
};

export default ApplyModal;
