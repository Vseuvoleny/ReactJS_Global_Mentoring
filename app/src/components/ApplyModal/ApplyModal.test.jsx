import React from "react";
import { render } from "@testing-library/react";
import ApplyModal from "./ApplyModal";

jest.mock("../MovieForm/MovieForm", () => () => <div>Movie Form</div>);

describe("ApplyModal component", () => {
  test("should return MovieForm", () => {
    const { container, getByText } = render(<ApplyModal />);
    expect(container.textContent).toMatch("Movie Form");
    expect(getByText("add movie")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
