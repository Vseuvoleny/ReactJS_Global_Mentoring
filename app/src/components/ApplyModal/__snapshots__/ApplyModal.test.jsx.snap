// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplyModal component should return MovieForm 1`] = `
<div>
  <section
    class="apply-modal"
  >
    <span
      class="apply-modal__title"
    >
      add movie
    </span>
    <div>
      Movie Form
    </div>
  </section>
</div>
`;
